packages/next/build/babel/preset.ts

```diff
@@ -14,15 +14,13 @@ type StyledJsxBabelOptions =
 
 // Resolve styled-jsx plugins
 function styledJsxOptions(options: StyledJsxBabelOptions) {
-  if (!options) {
-    return {}
-  }
+  const baseOptions = { styleModule: 'next/styled-jsx/style', ...options }
```

Comment:
This can just be a `require.resolve` it seems.
```suggestion
  const baseOptions = { styleModule: require.resolve('styled-jsx/style'), ...options }
```

Reply:
I agree but it needs to be gated behind a `babel-loader` check - ref https://github.com/vercel/next.js/pull/19538

Reply:
@timneutkens tried that https://github.com/vercel/next.js/pull/21320/commits/1c9ad4dda84c8ce1edba8e6ea6b4422f3bb88e34. But then [something else](https://github.com/vercel/next.js/runs/1729313185) broke, it seems that the change caused CSS from styled-jsx not to be generated. Not sure if this is related to the issue @merceyz mentions

Reply:
Also note that even if we could use `require.resolve('styled-jsx/style')`, it would still be necessary to export `next/styled-jsx`, given that the project that depends on next.js won't be able to import `styled-jsx` directly.